### ISSUE TITLE
error_reporting detection

### ISSUE THREAD

**Issue by Rarst** [WordPress-Coding-Standards/WordPress-Coding-Standards#682]
```
FILE: ...p-content\plugins\wordpress-seo\admin\views\tool-file-editor.php
----------------------------------------------------------------------
FOUND 2 ERRORS AFFECTING 1 LINE
----------------------------------------------------------------------
 23 | ERROR | error_reporting is prohibited, changing configuration at
    |       | runtime is not allowed on VIP Production.
    |       | (WordPress.VIP.RestrictedFunctions.runtime_configuration)
 23 | ERROR | error_reporting is prohibited as it can lead to full path
    |       | disclosure.
    |       | (WordPress.VIP.RestrictedFunctions.prevent_path_disclosure)
----------------------------------------------------------------------
```

1. Duplication, two sniffs for same thing.
2. In this case it was `error_reporting( 0 )` which _disables_ it. Seems like this case could be skipped, if technically possible?

**Comment by jrfnl** [comment]
My two pennies:

> 1.   Duplication, two sniffs for same thing.

The duplication is not an issue in itself as people can modularly exclude a certain (part of) a sniff. If they exclude the one, this doesn't necessarily mean that the other reason why the function is being reporting should be disregarded too.

What we could do is break off processing for the token after it has been determined it's an error, so you wouldn't get the two notices, just the one. However, that does not play nice with the modular disabling of the partial sniffs as, in that case, if the first one would be disabled, the second wouldn't show anymore either as the exclusion of the error is done in PHPCS itself not in WPCS.

> 2.   In this case it was error_reporting( 0 ) which disables it. Seems like this case could be skipped, if technically possible?

This is a VIP sniff and AFAIK on the VIP platform, `error_reporting()` should never be needed/used.

**Comment by Rarst** [comment]
I suppose for a VIP rule this isn't worth trouble then. Thanks for looking!
